### ISSUE TITLE
Scheduler service keeps crashing

### ISSUE THREAD

**Issue by eiabea** [zammad/zammad-docker-compose#138]
### Infos:

* Docker version: 18.03.1-ce
* Docker-compose version: 1.19.0 (python package)
* Operating system (Docker host): Ubuntu 16.04.4 LTS (Kernel 4.4.0)

### Expected behavior:

* Running scheduler service

### Actual behavior:

Scheduler keeps crashing with following exception:

First some issues about the database connection

```
E, [2020-02-27T11:13:45.303285 #1-47406482021720] ERROR -- : thread_client 6337427470 exited with error #<ActiveRecord::ConnectionTimeoutError: could not obtain a connection from the pool within 5.000 seconds (waited 5.000 seconds); all pooled connections were in use>
E, [2020-02-27T11:13:45.303628 #1-47406482021720] ERROR -- : /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:201:in `block in wait_poll'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:190:in `loop'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:190:in `wait_poll'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:151:in `internal_poll'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:277:in `internal_poll'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:145:in `block in poll'
  /usr/local/lib/ruby/2.5.0/monitor.rb:226:in `mon_synchronize'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:155:in `synchronize'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:145:in `poll'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:818:in `acquire_connection'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:538:in `checkout'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:382:in `connection'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:1033:in `retrieve_connection'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_handling.rb:118:in `retrieve_connection'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/connection_handling.rb:90:in `connection'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/relation/calculations.rb:197:in `block in pluck'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/relation.rb:584:in `skip_query_cache_if_necessary'
  /usr/local/bundle/gems/activerecord-5.2.4.1/lib/active_record/relation/calculations.rb:197:in `pluck'
  /opt/zammad/app/models/ticket/overviews.rb:25:in `all'
  /opt/zammad/app/models/ticket/overviews.rb:96:in `index'
  /opt/zammad/lib/sessions/backend/ticket_overview_list.rb:47:in `load'
  /opt/zammad/lib/sessions/backend/ticket_overview_list.rb:109:in `push'
  /opt/zammad/lib/sessions/client.rb:69:in `each'
  /opt/zammad/lib/sessions/client.rb:69:in `block in fetch'
  /opt/zammad/lib/sessions/client.rb:25:in `loop'
  /opt/zammad/lib/sessions/client.rb:25:in `fetch'
  /opt/zammad/lib/sessions/client.rb:7:in `initialize'
  /opt/zammad/lib/sessions.rb:721:in `new'
  /opt/zammad/lib/sessions.rb:721:in `thread_client'
  /opt/zammad/lib/sessions.rb:676:in `block (3 levels) in jobs'
  /usr/local/bundle/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
```
These issues don't kill the service, but after some occurrences the service crashes with:

```
#<Thread:0x0000563b63342750@/usr/local/bundle/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:471 run> terminated with exception (report_on_exception is true):
/opt/zammad/lib/sessions.rb:745:in `rescue in thread_client': STOP thread_client for client .9316613981 after 10 tries (RuntimeError)
	from /opt/zammad/lib/sessions.rb:720:in `thread_client'
	from /opt/zammad/lib/sessions.rb:676:in `block (3 levels) in jobs'
	from /usr/local/bundle/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
RuntimeError: STOP thread_client for client .9316613981 after 10 tries
  /opt/zammad/lib/sessions.rb:745:in `rescue in thread_client'
  /opt/zammad/lib/sessions.rb:720:in `thread_client'
  /opt/zammad/lib/sessions.rb:676:in `block (3 levels) in jobs'
  /usr/local/bundle/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
bundler: failed to load command: script/scheduler.rb (script/scheduler.rb)
I, [2020-02-27T11:13:53.216359 #1-47406448813500]  INFO -- : Scheduler stopped.
```
The container gets restarted and everything starts over again.

I already tried to set the `max_connections` of the postgres to `500` without any luck

### Steps to reproduce the behavior:

Fresh installation of Zammad using the docker-compose file and the Zammad images with the version `3.2.0-13`

**Comment by monotek** [comment]
can't reproduce.

current ci test looks good too:  https://github.com/zammad/zammad-docker-compose/runs/471973598?check_suite_focus=true

**Comment by eiabea** [comment]
@monotek as far as i understand the [test.sh](https://github.com/zammad/zammad-docker-compose/blob/master/.github/tests.sh) script is that it only waits for the service to come up on port 80, which is the `railserver`, but i reported that the scheduler is crashing.

**Comment by monotek** [comment]
if you have a look at the log you see the scheduler works...

**Comment by eiabea** [comment]
Indeed, but the container is only running for about 6 seconds. The issues occur after about 30 seconds.

I will try to deploy zammad on a different host and report back if the issues persist

**Comment by monotek** [comment]
I've tried several times locally in the last days without problems.

**Comment by Cybernisk** [comment]
@eiabea I have no solution but have the same issue. See the last message [here](https://community.zammad.org/t/found-numerous-zammad-issues/3756/8). 
TL;DR: In my case when scheduler hits 50 connections to the DB it fails with an error `could not obtain a connection from the pool within 5.000 seconds (waited 5.000 seconds); all pooled connections were in use`

**Comment by MrGeneration** [comment]
You could raise die maximum allowed number of connections within postgres:
https://hub.docker.com/_/postgres/#!

Search for `max_connections`. Setting this to 500 should be far enough.
This is out of scope of our docker context I'm afraid, because postgres doesn't provide environment variables for this as of now.

**Comment by Cybernisk** [comment]
Unfortunately,  I've already tried max_connections = 500 (default for Zammad's Postgres build is 300 for now). But the issue is not with Postgress, because when I said "hits 50 connections" I meant **only** scheduler connections (see logs [here](https://community.zammad.org/t/found-numerous-zammad-issues/3756/8)). The sum of all DB connections is not over 100 for my case.
